install/scripts/backup-pit-data.sh

```diff
@@ -0,0 +1,119 @@
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+set -euo pipefail
+
+# This means that something like /tmp/*.log will evaluate to an empty string if no files fit the pattern
+shopt -s nullglob
+
+# This script is a replacement for the steps that were previously done manually
+# during the "Deploy Final NCN" step of CSM installs.
+
+function err_exit {
+  echo "ERROR: $*" >&2
+  exit 1
+}
+
+function dir_exists {
+  [[ -e $1 ]] || err_exit "Directory '$1' does not exist"
+  [[ -d $1 ]] || err_exit "'$1' exists but is not a directory"
+}
+
+function run_cmd {
+  echo "# $*"
+  "$@" || err_exit "Command failed with exit code $?: $*"
+}
+
+# Ensure that PITDATA and CSM_PATH variables are set
+[[ -v PITDATA && -n ${PITDATA} ]] || err_exit "PITDATA variable must be set"
+[[ -v CSM_PATH && -n ${CSM_PATH} ]] || err_exit "CSM_PATH variable must be set"
+
+# Make sure that expected directories exist and are actually directories
+for DIR in "${PITDATA}" "${PITDATA}/prep" "${PITDATA}/configs" "${CSM_PATH}" \
+           "${CSM_PATH}/images" "${CSM_PATH}/images/pre-install-toolkit"
+do
+  dir_exists "${DIR}"
+done
+
+PIT_ISO_DIR="${CSM_PATH}/images/pre-install-toolkit"
+
+# Make sure that expected PIT iso file can be found
+compgen -G "${PIT_ISO_DIR}/pre-install-toolkit*.iso" >/dev/null 2>&1 || err_exit "PIT ISO file (${PIT_ISO_DIR}/pre-install-toolkit*.iso) not found"
+
+# Okay, everything seems good
+run_cmd mkdir -pv "${PITDATA}"/prep/logs
+
+# Because some of these files are log files that are changing during this procedure, any call to directly
+# tar them may result in the tar command failing. Thus, we first copy all of these files into a temporary
+# directory, and from there we create the tar archive
+
+TEMPDIR=$(mktemp -d) || err_exit "Command failed: mktemp -d"
+
+echo "Copying selected files to temporary directory"
+
+for BACKUP_TARGET in \
+    /etc/conman.conf \
+    /etc/dnsmasq.d \
+    /etc/os-release \
+    /etc/sysconfig/network \
+    /opt/cray/tests/cmsdev.log \
+    /opt/cray/tests/install/logs \
+    /opt/cray/tests/logs \
+    /root/.bash_history \
+    /root/.canu \
+    /root/.config/cray/logs \
+    /root/csm*.{log,txt} \
+    /tmp/*.log \
+    /usr/share/doc/csm/install/scripts/csm_services/yapl.log \
+    /var/log
+do
+  [[ -e ${BACKUP_TARGET} ]] || continue
+  DIRNAME=$(dirname "${BACKUP_TARGET}")
+  TARG_DIR="${TEMPDIR}${DIRNAME}"
+  run_cmd mkdir -pv "${TARG_DIR}"
+  run_cmd cp -pr "${BACKUP_TARGET}" "${TARG_DIR}"
+done
+
+echo "Creating PIT backup tarfile"
+
+pushd "${TEMPDIR}"
+run_cmd tar -czvf "${PITDATA}/prep/logs/pit-backup-$(date +%Y-%m-%d_%H-%M-%S).tgz" --remove-files *
+popd
+run_cmd rmdir -v "${TEMPDIR}"
+
+echo "Copying files to ncn-m002"
+ssh ncn-m002 \
```

Comment:
Since we're fixing this up, should we also set this to find the first-master-hostname instead of hardcoding `ncn-m002` in? This would help in case a system had some reason that m002 was unavailable or if they used a different first master.

Reply:
Sure thing. I'll also need to modify the corresponding section that copies it back, but should be easy enough

Reply:
Done!